packages/optimizers/terser/src/TerserOptimizer.js

```diff
@@ -90,11 +90,11 @@ export default new Optimizer({
     let minifiedContents: string = nullthrows(result.code);
     if (result.map && typeof result.map !== 'string') {
       sourceMap = new SourceMap();
-      sourceMap.addRawMappings(
-        result.map.mappings,
-        result.map.sources,
-        result.map.names || [],
-      );
+      sourceMap.addRawMappings({
```

Comment:
So in `loadSourceMap` you did `sourceMap.addRawMapping(map)` and here it's only using these 4 properties. I think should make one of these the preferred way (and maybe document that for plugin authors)?

Reply:
preferred way is `sourceMap.addRawMappings(map)` but apparently the types for Terser are read-only or something, so it wasn't possible here...

Didn't even know read-only really existed in JS

Reply:
I'll definitely write a bunch of sourcemap docs at some point

Reply:
> Didn't even know read-only really existed in JS and pretty sure Terser does not output read-only object

It's the covariant type annotation on the object property in flow (`+file: string`).

Doing this in the sourcemap packages resolves that:
```
export type VLQMap = {
  +sources: Array<string>,
  +sourcesContent?: Array<string | null>,
  +names: Array<string>,
  +mappings: string,
  +version?: number,
  +file?: string,
  +sourceRoot?: string,
  ...
};
```

alternatively we could remove the `+` from RawSourceMap in `flow-libs/terser.js.flow`

Reply:
Probably makes sense to do it, we don't really set anything anyway